src/server/api/controllers/messages-controller.js

```diff
@@ -1,10 +1,48 @@
+/* eslint-disable @typescript-eslint/camelcase */
+/* eslint-disable camelcase */
 const knex = require('../../config/db');
 const Error = require('../lib/utils/http-error');
 const moment = require('moment-timezone');
 
-const getMessages = async () => {
+// getting all messages from channel_messages and query based on query, channel_id, sender, limit, sort and date
+const getMessages = async (query, channel_id, sender, limit, sort, date) => {
+  const get_channel_messages = knex('channel_messages').distinct('*');
   try {
-    return await knex('messages').select('messages.id', 'messages.title');
+    if (query) {
```

Comment:
This prevents us from having query string `query=something&sender=John&sort=desc`. Did you think about that scenario?

Reply:
Not exactly . I did based on task and they are one by one. I will check again .

Reply:
Hi Please check again !

Reply:
You are correct that the ticket said that. Just wondered if you thought this was interesting enough to do.